daemon/cmd/daemon.go

```diff
@@ -414,6 +414,18 @@ func NewDaemon(ctx context.Context, dp datapath.Datapath) (*Daemon, *endpointRes
 			log.Info("k8s mode: Allowing localhost to reach local endpoints")
 		}
 
+		// Now that we know the node nodeIP of the node, we can get the mtu of the interface that holds that IP
+		if node.GetExternalIPv4() != nil {
+			mtuConfig = mtu.NewConfiguration(authKeySize, option.Config.EnableIPSec, option.Config.Tunnel != option.TunnelDisabled, configuredMTU, node.GetExternalIPv4())
```

Comment:
This seems incredibly fragile as the old MTU has already been passed into the NodeDiscovery layer and thus been pushed down to the datapath layer. We need to find a better way. This will create very subtle bugs in the future if we change the MTU suddenly underneath after some of the initialization has already happened.

Reply:
That's right. It seems that we need to move the whole section:

```
bootstrapStats.k8sInit.Start()
[...]
bootstrapStats.cleanup.Start()
[...]
if k8s.IsEnabled() {
[...]
}
```

before we call `nodediscovery.NewNodeDiscovery()` and before we create an instance of `Daemon` struct.

@manuelbuil can you try that? You will need possibly to figure out whether `bootstrapStats.k8sInit.Start()` can be done before the first call of `mtu.NewConfiguration` and move up other variables which might be need for that.

Reply:
Yes, I am planning to look into this as soon as I have some time. I am currently busy with some urgent stuff. Hopefully by the end of this week I'll have some time :)

Reply:
@mrostecki I tried what you suggested and it is possible to move code before the daemon creation. At least I haven't found any problem by manual testing and now the correct mtu is calculated before the daemon is created. I'll update the new version + rebase in a minute